File: Python/Biggners/Number_Guessing_Game/Project.py
import random
import math

# Initializing the game
def game():

    # Taking Inputs
    lower_bound = int(input("Enter Lower bound :- "))
    upper_bound = int(input("Enter Upper bound :- "))

    # generating a random number between the lower and upper bound
    chances = 0
    random_number = random.randint(lower_bound, upper_bound)
    chances = round(math.log(upper_bound - lower_bound + 1, 2))
    print("\n\tYou've only ", chances, " chances to guess the integer!\n")

    # Initializing the number of guesses.

    guess_count = 0

    while guess_count < math.log(upper_bound - lower_bound + 1, 2):
        guess_count += 1
        if guess_count < chances:
                print("You still have ", (chances + 1) - guess_count, " chances to guess the number")
        elif guess_count == chances:
            print("This is your last chance!")
        else:
            print("Your chances are over!")
        # Taking guessing number as input

        guess = int(input("Guess a number :- "))

        if random_number == guess:
            print("Congratulations you did it in ", guess_count, " try")
            print("Want to play again? (yes/no)")
            ans = input("Yes/No :- ")
            if ans == "Yes" or ans == "yes":
                game()
            elif ans == "No" or ans == "no":
                print("Thanks for playing!")
                break
            else:
                print("Invalid Input")
                break 

        elif random_number > guess:
            print("You guessed too small!")
            print("Try Again")

        else:
            print("You Guessed too high!")
            print("Try Again")

        # If Guessing is more than required guesses
        if guess_count >= math.log(upper_bound - lower_bound + 1, 2):
            print("\nThe number is %d" % random_number)
            print("\tBetter Luck Next time!")
            break

# Driver Code
if __name__ == "__main__":
    # Calling the game function
    game()

File: C/Biggners/Number_guessing_game/Project.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int GUESS(){
    int number, guess, nguesses=1;
    srand(time(0));

    // Generate a random number between 1 and 100
    number = rand()%100 + 1;

    do{
        printf("Guess the number between 1 to 100\n");
        scanf("%d", &guess);
        if (guess > number){
            printf("You guessed too high\n");
            printf("Guess again\n");
        }
        else if(guess < number){
            printf("You guessed too low\n");
            printf("Guess again\n");
        }
        else{
            printf("You guessed the number in %d att    empts\n", nguesses);
            printf("Congratulations\n");
        }
        nguesses++;
    }
    while (guess != number);
    return 0;
}